test/fuzz/transformation_make_vector_operation_dynamic_test.cpp

```diff
@@ -0,0 +1,328 @@
+
+  transformation = TransformationMakeVectorOperationDynamic(38, 11);
+  ASSERT_TRUE(
+      transformation.IsApplicable(context.get(), transformation_context));
+
+  transformation = TransformationMakeVectorOperationDynamic(39, 10);
+  ASSERT_TRUE(
+      transformation.IsApplicable(context.get(), transformation_context));
+}
+
+TEST(TransformationMakeVectorOperationDynamicTest, Apply) {
+  std::string reference_shader = R"(
+               OpCapability Shader
+          %1 = OpExtInstImport "GLSL.std.450"
+               OpMemoryModel Logical GLSL450
+               OpEntryPoint Vertex %20 "main"
+
+; Types
+          %2 = OpTypeVoid
+          %3 = OpTypeFunction %2
+          %4 = OpTypeInt 32 0
+          %5 = OpTypeFloat 32
+          %6 = OpTypeVector %5 2
+          %7 = OpTypeVector %5 3
+          %8 = OpTypeVector %5 4
+
+; Constant scalars
+          %9 = OpConstant %4 0
+         %10 = OpConstant %4 1
+         %11 = OpConstant %4 2
+         %12 = OpConstant %4 3
+         %13 = OpConstant %5 0
+         %14 = OpConstant %5 1
+         %15 = OpConstant %5 2
+         %16 = OpConstant %5 3
+
+; Constant vectors
+         %17 = OpConstantComposite %6 %13 %14
+         %18 = OpConstantComposite %7 %13 %14 %15
+         %19 = OpConstantComposite %8 %13 %14 %15 %16
+
+; main function
+         %20 = OpFunction %2 None %3
+         %21 = OpLabel
+         %22 = OpCompositeExtract %5 %17 0
+         %23 = OpCompositeExtract %5 %17 1
+         %24 = OpCompositeExtract %5 %18 0
+         %25 = OpCompositeExtract %5 %18 1
+         %26 = OpCompositeExtract %5 %18 2
+         %27 = OpCompositeExtract %5 %19 0
+         %28 = OpCompositeExtract %5 %19 1
+         %29 = OpCompositeExtract %5 %19 2
+         %30 = OpCompositeExtract %5 %19 3
+         %31 = OpCompositeInsert %6 %13 %17 0
+         %32 = OpCompositeInsert %6 %14 %17 1
+         %33 = OpCompositeInsert %7 %13 %18 0
+         %34 = OpCompositeInsert %7 %14 %18 1
+         %35 = OpCompositeInsert %7 %15 %18 2
+         %36 = OpCompositeInsert %8 %13 %19 0
+         %37 = OpCompositeInsert %8 %14 %19 1
+         %38 = OpCompositeInsert %8 %15 %19 2
+         %39 = OpCompositeInsert %8 %16 %19 3
+               OpReturn
+               OpFunctionEnd
+  )";
+
+  const auto env = SPV_ENV_UNIVERSAL_1_5;
+  const auto consumer = nullptr;
+  const auto context =
+      BuildModule(env, consumer, reference_shader, kFuzzAssembleOption);
+  ASSERT_TRUE(IsValid(env, context.get()));
+
+  FactManager fact_manager;
+  spvtools::ValidatorOptions validator_options;
+  TransformationContext transformation_context(&fact_manager,
+                                               validator_options);
+
+  auto transformation = TransformationMakeVectorOperationDynamic(22, 9);
+  transformation.Apply(context.get(), &transformation_context);
+
+  transformation = TransformationMakeVectorOperationDynamic(23, 10);
+  transformation.Apply(context.get(), &transformation_context);
```

Comment:
I think it probably makes sense to ASSERT that each transformation is applicable. That way, we might get a test failure sooner, closer to the actual issue.

Reply:
Done.